### PR TITLE
Fix min qty in front ProductController

### LINKED ISSUE #35888: FO - minimum qty input on products listing, product detail pages and quick view
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Related to https://github.com/PrestaShop/hummingbird/pull/609

The minimum quantity for sale is never updated even after I added correct quantity of the product to my cart.

https://github.com/PrestaShop/PrestaShop/assets/16019289/5c5b0065-7c45-4377-a065-0082fb68ca57

This issue a reproduced on `8.1.x` with classic theme as well as on `develop` with hummingbird theme.

As discussed with @boherm , this should be fixed directly on the core, not on the theme.

### Expected behavior

Once the minimum quantity for sale of a product is reached in the cart, the minimum quantity for sale should be updated to 1

### Steps to reproduce

1. Go to BO > Edit a product > Edit the Minimum quantity for sale = 3
2. Go to FO > Quick view on the product
3. Add 3 products to cart > Continue Shopping
4. Click Quick view again > See quantity is still 3, it should be 1 :x:
5. Click on proceed to checkout
6. Open the product detail page
7. See in quantity = 3 :x: should be 1 as I already have the product in my cart

### PrestaShop version(s) where the bug happened

8.1.x, develop

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by boherm**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Fix min qty in front ProductController on ajax refresh action.<br>Now, we refreshing correctly the minimal required quantities on standard, virtual, combinaison and pack products.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See #35888 
| UI Tests          | https://github.com/boherm/ga.tests.ui.pr/actions/runs/8651056510 https://github.com/florine2623/testing_pr/actions/runs/8749975555 ✅ 
| Fixed issue or discussion?     | Fixes #35888
| Related PRs       | https://github.com/PrestaShop/hummingbird/pull/609
| Sponsor company   | PrestaShop SA

### CODE REVIEW

**Review by kpodemski — CHANGES_REQUESTED** [review]
I blocking it because i think this solution hurts performance, i will provide you with more details tomorrow! 👍

**Comment by Hlavtox** [comment]
I think that this should be fixed on ProductLazyArray level, so all places where the product is displayed have the same data.

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -1511,6 +1517,32 @@ private function getProductPageTitle(array $meta = null)
...
     }
 
+    /**
+     * Get product quantity currently in cart.
+     *
+     * @param ProductLazyArray $product
+     *
+     * @return int
+     */
+    private function getProductQuantityInCart(ProductLazyArray $product)
```
Comment (Hlavtox) [inline controllers/front/ProductController.php]: On my phone rn, but isnt there a function that does this?

Comment (boherm) [reply controllers/front/ProductController.php]: Yeah, `getProductQuantity` in `cart` class!
I will use that instead of my custom function ;)

Comment (matks) [reply controllers/front/ProductController.php]: Reviewing on the phone? Sounds fun!

**Review by matks — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @boherm ,

Tested on 8.1.x with standard, combination, virtual and pack product !

It doesn't work as expected :/

Steps to reproduce :
1. In BO > Product page > Change minimum quantity to sell to 5
2. Open product page in FO -> See minimum quantity is 5 ✔️ 
3. Add 5 products to the cart 
See quantity input is 1 ✔️ 
See message "The minimum purchase order quantity for the product is 5." is still displayed :x:
4. Refresh page -> Quantity input is back to 5 :x:


https://github.com/PrestaShop/PrestaShop/assets/16019289/b9f76fe4-be77-4c71-838b-918ab361344d




Could you check please ? :D 
Thanks!

**Comment by florine2623** [comment]
@kpodemski is it okay for you ? could you give your feedback ? :)

**Review by kpodemski — APPROVED** [review]

File: controllers/front/ProductController.php (hunk 1 above)
Comment (Hlavtox) [reply controllers/front/ProductController.php]: What if you directly call $this->context->cart->getProductQuantity($product->getId(), $product['id_product_attribute'])['quantity']?

**Review by kpodemski — CHANGES_REQUESTED** [review]
I'll revert my approval because @florine2623 reported some issues and I think they are relevant.

**Comment by boherm** [comment]
> Hello @boherm ,
> 
> Tested on 8.1.x with standard, combination, virtual and pack product !
> 
> It doesn't work as expected :/
> 
> Steps to reproduce :
> 
> 1. In BO > Product page > Change minimum quantity to sell to 5
> 2. Open product page in FO -> See minimum quantity is 5 ✔️
> 3. Add 5 products to the cart
>    See quantity input is 1 ✔️
>    See message "The minimum purchase order quantity for the product is 5." is still displayed ❌
> 4. Refresh page -> Quantity input is back to 5 ❌
> 
>  Screen.Recording.2024-04-19.at.10.42.21.mov 
> Could you check please ? :D Thanks!

Thanks for your test @florine2623, but i'm not totally agree with you about the "The minimum purchase order quantity for the product is 5.". In fact, I think that this message must be displayed always because the minimum purchase order quantity for the product are always 5, even if we have already 5 products in the cart. Maybe I'm wrong about this message 🤔

About the refresh issue, I can't reproduce on my side 😢

**Review by florine2623 — COMMENTED** [review]
Hello @boherm ,

Indeed, when I re-read my message, I can't understand why I wrote that the minimal quantity message is wrong. It is normal that it is always displayed ^^

In this screen record, you can see that when I already have the minimal quantity of purchase in my cart (10 for example), if I refresh my page, I can't add just 1 more product to my cart.

https://github.com/PrestaShop/PrestaShop/assets/16019289/817efcbd-d68c-4dee-9aae-28463dacf41d

I followed the same steps to reproduce that in the previous message.

> In BO > Product page > Change minimum quantity to sell to 5
> Open product page in FO -> See minimum quantity is 5 ✔️
> Add 5 products to the cart
> See quantity input is 1 ✔️
> See message "The minimum purchase order quantity for the product is 5." is still displayed ❌
> Refresh page -> Quantity input is back to 5 ❌

**Review by jolelievre — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @boherm ,

Tested with all types of products products.

Tested after a refresh, with/without the item(s) in the cart, with/without different minimum quantities for each combination, on quickview modal.

Works exactly as expected !
It is QA ✅ 
Thanks!

**Pull request merged**